packages/core/src/fhir-deduplication/__tests__/group-same-diag-reports.test.ts

```diff
@@ -12,31 +12,68 @@ let diagReportId: string;
 let diagReportId2: string;
 let diagReport: DiagnosticReport;
 let diagReport2: DiagnosticReport;
+let practId: string;
+let practId2: string;
+let practRef: Reference<Practitioner>;
+let practRef2: Reference<Practitioner>;
 
 beforeEach(() => {
+  practId = faker.string.uuid();
+  practId2 = faker.string.uuid();
+
+  practRef = { reference: `Practitioner/${practId}` };
+  practRef2 = { reference: `Practitioner/${practId2}` };
+
   diagReportId = faker.string.uuid();
   diagReportId2 = faker.string.uuid();
-  diagReport = makeDiagnosticReport({ id: diagReportId });
-  diagReport2 = makeDiagnosticReport({ id: diagReportId2 });
+  diagReport = makeDiagnosticReport({ id: diagReportId, result: resultExample });
+  diagReport2 = makeDiagnosticReport({ id: diagReportId2, presentedForm: presentedFormExample });
 });
 
 describe("groupSameDiagnosticReports", () => {
-  it("correctly groups duplicate diagReports based on effectiveDateTime and data presence", () => {
+  it("groups diagReports with the same effectiveDateTime and performer ref", () => {
     diagReport.effectiveDateTime = dateTime.start;
     diagReport2.effectiveDateTime = dateTime.start;
-    diagReport.result = resultExample;
-    diagReport2.presentedForm = presentedFormExample;
+    diagReport.performer = [practRef];
+    diagReport2.performer = [practRef];
+
+    const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
+    expect(diagReportsMap.size).toBe(1);
+  });
+
+  it("groups diagReports with the same effectiveDateTime, if one of them has a performer", () => {
+    diagReport.effectiveDateTime = dateTime.start;
+    diagReport2.effectiveDateTime = dateTime.start;
+    diagReport.performer = [practRef];
 
     const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
     expect(diagReportsMap.size).toBe(1);
   });
 
+  it("does not group diagReports with the same effectiveDateTime if none of them have a practitioner reference ", () => {
+    diagReport.effectiveDateTime = dateTime.start;
+    diagReport2.effectiveDateTime = dateTime.start;
+
+    const { diagReportsMap } = groupSameDiagnosticReports([diagReport, diagReport2]);
+    expect(diagReportsMap.size).toBe(2);
+  });
```

Comment:
That's weird, no? Group if same date and one reference, but not if same date and no reference?

Reply:
Yea, good point. I changed the logic to be like this:
- same date same performer => match
- same date no performer => match
- same date, one with, one without performer => no match